download-maven-plugin/src/main/java/com/googlecode/WGet.java

```diff
@@ -215,12 +221,46 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     // DO
     try
     {
-      if (outputFile.exists() && !overwrite)
+      boolean haveFile = outputFile.exists();
+      if (haveFile)
       {
+        if (this.checkSignature && this.md5 != null)
+        {
+          try {
+            SignatureUtils.verifySignature(outputFile, this.md5,
+                    MessageDigest.getInstance("MD5"));
+          } catch (MojoFailureException e) {
+            getLog().warn("File already exist. " + e.getMessage());
```

Comment:
done, here I also improved the code removing some dups and using a more clear (boolean) variable to perform the checks.